tests/coalaJSONTest.py

```diff
@@ -87,42 +96,45 @@ def test_show_bears_attributes(self):
                              )
             self.assertTrue(bear['metadata']['optional_params'])
             self.assertFalse(bear['metadata']['non_optional_params'])
+            self.assertFalse(stderr)
 
     @unittest.mock.patch('coalib.parsing.DefaultArgParser.get_all_bears_names')
     @unittest.mock.patch('coalib.collecting.Collectors.icollect_bears')
     def test_version_conflict_in_collecting_bears(self, import_fn, _):
         with bear_test_module():
             import_fn.side_effect = VersionConflict('msg1', 'msg2')
-            retval, _ = execute_coala(coala.main, 'coala', '--json', '-B')
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '--json', '-B')
             self.assertEqual(retval, 13)
 
     def test_text_logs(self):
-        retval, output = execute_coala(
+        retval, stdout, stderr = execute_coala(
             coala.main, 'coala', '--json', '-c', 'nonex')
+        test_text = '{\n  "results": {}\n}\n'
         self.assertRegex(
-            output,
+            stderr,
             ".*\\[ERROR\\].*The requested coafile '.*' does not exist. .+\n")
+        self.assertEqual(stdout, test_text)
 
     def test_output_file(self):
         with prepare_file(['#todo this is todo'], None) as (lines, filename):
-            retval, output = execute_coala(coala.main, 'coala', '--json',
-                                           '-c', os.devnull,
-                                           '-b', 'LineCountTestBear',
-                                           '-f', re.escape(filename),
-                                           '--log-json',
-                                           stdout_only=True)
+            retval, stdout, stderr = execute_coala(coala.main, 'coala',
+                                                   '--json', '-c', os.devnull,
+                                                   '-b', 'LineCountTestBear',
+                                                   '-f', re.escape(filename),
+                                                   '--log-json')
             execute_coala(coala.main, 'coala', '--json', '-c', os.devnull,
```

Comment:
put the args into a tuple, and then call `execute_coala` twice with the `*<tuple>` , so it is easier to see what is occurring here.

But still the returned values of this execution should be checked.

`retval2, stdout2, stderr2 = ...`